[Kuan Min] iP

Rikko1204 opened the pull request:
# AronaPro

> Sensei, what can I do for you today?

AronaPro frees your mind of having to remember things you need to do. It's,
- Text-based
- Easy to learn
- ~~FAST~~ *SUPER* FAST to use

All you need to do is,

1. Download it from [here](https://github.com/Rikko1204/ip/releases/tag/v1.0.0).
1. Double-click it.
1. Add your tasks.
1. Let it manage your tasks for you 😃 

And it is **FREE!**

Features:

- [X] Managing tasks
- [X] Managing deadlines
- [ ] Reminders (coming soon)

If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:
```java
    public static void main(String[] args) throws TaskException {
        Arona arona = new Arona();
        arona.run();
    }
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

Rikko1204 committed:
Chatbot skeleton

Rikko1204 committed:
Added echo commands

Rikko1204 committed:
Added task lists

Rikko1204 committed:
Added task done

Rikko1204 committed:
Added task types

Rikko1204 committed:
Added exception handling

Rikko1204 committed:
Added task delete

Rikko1204 committed:
Added testings

Rikko1204 committed:
Change chatbot name to Arona

Rikko1204 committed:
Add load and save file

Rikko1204 committed:
Add date

Rikko1204 committed:
Add OOP WIP

Rikko1204 committed:
Test

Rikko1204 committed:
Add more OOP (branch)

Rikko1204 committed:
Add more OOP

Rikko1204 committed:
Add packages

Rikko1204 committed:
Merge branch 'add-gradle-support' of https://github.com/Rikko1204/ip

Rikko1204 committed:
Add JUnit tests

Rikko1204 committed:
Add find

Rikko1204 committed:
add JavaDoc

Rikko1204 committed:
Merge branch 'branch-A-JavaDoc'

Rikko1204 committed:
Improve coding quality

Rikko1204 committed:
Fix conflict

Rikko1204 committed:
Add coding standards

Rikko1204 committed:
Add small code changes

legionlegion commented on src/main/java/arona/ListTask.java:
```diff
@@ -0,0 +1,25 @@
+     * @param taskList The class that contains the tasklist
+     * @param ui UI that handles all the console output
+     * @param storage Handle the loading and saving of data
+     * @throws FileException if file cannot be found
```
Perhaps you should remove these thrown exceptions to match your function and avoid confusion?

legionlegion commented on src/main/java/arona/AronaException.java:
```diff
@@ -0,0 +1,7 @@
+package arona;
+
```
Perhaps it would be good to describe the purpose of this class because it seemed a bit redundant at first glance.

legionlegion commented on src/main/java/arona/Ui.java:
```diff
@@ -0,0 +1,68 @@
+            /*   */              " /    |    \\  | \\_ ( <_> )   |   \\/ __ \\_ \n" +
+            /* */                " \\____|__  /__|    \\____/|___|_  (____  / \n" +
+            /* */                "         \\/                    \\/     \\/ ";
+    private String greetings = "こんにちは先生、私はアロナです. \n"
```
I like your use of Japanese, but perhaps it could be confusing for some users to read?

legionlegion commented on src/main/java/arona/Ui.java:
```diff
@@ -0,0 +1,68 @@
+        }
+    }
+
+    public void taskAdded(ArrayList<Task> tasks) {
```
perhaps the name here can be a verb like printTaskAdded?

legionlegion reviewed (commented):
LGTM! I like the effort made to refactor the code into many different classes. Just some nits to fix!

Rikko1204 committed:
Add GUI

Joshy837 commented on src/main/java/arona/MainWindow.java:
```diff
@@ -0,0 +1,71 @@
+ * Controller for arona.MainWindow. Provides the layout for the other controls.
+ */
+public class MainWindow extends AnchorPane {
+    @FXML
```
I like how you hid the class information as "private", following the coding standard. 👍

Joshy837 reviewed (commented):
I like how you hid the class information as "private", following the coding standard. 👍 And also how you separated the fields into different types.

Joshy837 commented on src/main/java/arona/MainWindow.java:
```diff
@@ -0,0 +1,71 @@
+        scrollPane.vvalueProperty().bind(dialogContainer.heightProperty());
+    }
+
+    public void setArona(Arona arona) {
```
I like how you followed the camelCase formatting, following the coding standard.

Joshy837 reviewed (commented):
I like how you followed the camelCase formatting, following the coding standard.

Joshy837 commented on src/main/java/arona/TaskList.java:
```diff
@@ -0,0 +1,129 @@
+    public ArrayList<Task> getTasks() {
+        return tasks;
+    }
+    public void addTask(String input) throws TaskException {
```
Can leave a spacing after the above method to follow the coding standard.

Joshy837 reviewed (commented):
Can leave a spacing after the above method to follow the coding standard.

Rikko1204 committed:
Update jar application

Rikko1204 committed:
Add assertions

Add assertions to multiple methods in different classes to test for
assumptions.

There are no assertions to determine if there is an error in programming
logic.

Adding assertions allows us to verify the correctness of the programming
logic. We will be able to identify the error early and prevent the
effect from cascading further.

Let's add assertions to multiple methods in different classes to test
our assumptions.

Rikko1204 committed:
Add code quality improvements

There are some code quality improvements that can be made.

Maintaining code quality is important for preserving code readability
and extensibility.

Improve quality of code where necessary by following the convention, not
limited to naming conventions, layouts and comments.

Let's improve the code quality by following the convention.

Rikko1204 committed:
Merge pull request #1 from Rikko1204/branch-A-Assertions

Add assertions

Rikko1204 committed:
Merge branch 'master' into branch-A-CodeQuality

# Conflicts:
#	src/data/tasklist.txt

Rikko1204 committed:
Merge pull request #2 from Rikko1204/branch-A-CodeQuality

Branch a code quality

Rikko1204 committed:
Add detect duplicate ability

Rikko1204 committed:
Merge branch 'branch-A-DetectDuplicates'

Rikko1204 committed:
Add product screenshot

Rikko1204 committed:
Update README.md

Rikko1204 committed:
Update README.md

Rikko1204 committed:
Add more javadoc

Rikko1204 committed:
Add edits

Rikko1204 committed:
Merge branch 'test'

Rikko1204 committed:
jar fix

Rikko1204 committed:
Merge branch 'jar-fix'

Rikko1204 committed:
Update README.md

Rikko1204 committed:
Update README.md

Rikko1204 committed:
Add User Guide

Rikko1204 committed:
Merge branch 'master' of https://github.com/Rikko1204/ip
